app/services/dttp/retrieve_providers.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Dttp
+  class RetrieveProviders
+    include ServicePattern
+
+    class HttpError < StandardError; end
+
+    MAX_PAGE_SIZE = 5000
+
+    HEADERS = {
+      "Prefer" => "odata.maxpagesize=#{MAX_PAGE_SIZE}",
+    }.freeze
+
+    INSTITUTION_TYPE_IDS = [
+      "b5ec33aa-216d-e711-80d2-005056ac45bb", # EBITT
+      "b7ec33aa-216d-e711-80d2-005056ac45bb", # EYITT
+      "b9ec33aa-216d-e711-80d2-005056ac45bb", # HEI
+      "bbec33aa-216d-e711-80d2-005056ac45bb", # ITT Provider - HESA
+      "bdec33aa-216d-e711-80d2-005056ac45bb", # ITT Provider - Non-HESA
+      "bfec33aa-216d-e711-80d2-005056ac45bb", # NonHEI (data gathered under Scitt process)
+      "c1ec33aa-216d-e711-80d2-005056ac45bb", # Non-HESA HEI
+    ].freeze
```

Comment:
can we move this to https://github.com/DFE-Digital/register-trainee-teacher-data/blob/master/app/lib/dttp/code_sets/

Reply:
done